### PR TITLE
add validation and test

### PR DESCRIPTION
**Pull request by akezeth**
Following tasks have been implemented

Create custom methods
Your models should include some additional methods.

Users
A method that returns the 3 most recent posts for a given user.
Posts
A method that updates the posts counter for a user.
A method which returns the 5 most recent comments for a given post.
Comments
A method that updates the comments counter for a post.
Likes
A method that updates the likes counter for a post.
Go to rails c and check if your methods are working.

### CODE REVIEW

File: spec/models/post_spec.rb [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+    expect(post.valid?).to eq false
+  end
+
+  it 'should not create post with like/comment counter not an integer' do
+    post = Post.create(author: User.create(name: 'Aklilu', posts_counter: 0), title: 'First Post',
+                       comments_counter: 'Hi', likes_counter: 'Hi')
+    expect(post.valid?).to eq false
+  end
+
+  it 'should create post with a title longer than 250 characters' do
```
Comment (muneebulrehman) [inline spec/models/post_spec.rb:21]: Kindly test for a scenario where the post length exceeds 250 characters. This is a part of the project requirements.

File: spec/models/post_spec.rb [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+    post = Post.create(author: User.create(name: 'Aklilu', posts_counter: 0), title: 'First Post',
+                       comments_counter: -1, likes_counter: -1)
+    expect(post.valid?).to eq false
+  end
+
+  it 'should not create post with like/comment counter not an integer' do
+    post = Post.create(author: User.create(name: 'Aklilu', posts_counter: 0), title: 'First Post',
+                       comments_counter: 'Hi', likes_counter: 'Hi')
+    expect(post.valid?).to eq false
+  end
```
Comment (muneebulrehman) [inline spec/models/post_spec.rb:19]: Kindly test for a scenario where both comments_counter and likes_counter are valid.

**Review by muneebulrehman — CHANGES_REQUESTED** [review]
Hi @akezeth , 

Good job so far! I see you put a lot of effort into this project. However, there are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights
- No linter errors
- Descriptive PR
- Good docs

## Required Changes ♻️
_Check the comments under the review._

- [ ] These tests are failing. Kindly fix the issue. You can resubmit for the review once all the tests are passed.
image


## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by MahmoudBakr23 — APPROVED** [review]
Hi @akezeth 😃, 

Your project is complete!  There is nothing else to say other than... It's time to merge it :shipit:
Congratulations! 🎉

**Highlights**
- Well done on making the required changes from the previous reviewer. 💯 

Cheers, and Happy coding!👏👏👏

Feel free to mention @MahmoudBakr23 to leave any questions or comments in the PR thread if something is not 100% clear.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**